python/ray/tune/syncer.py

```diff
@@ -1,1015 +1,33 @@
-        ):
-            return False
-
-        last_sync_time = self._sync_times.setdefault(trial.trial_id, float("-inf"))
-
-        if time.time() - last_sync_time < self._sync_period:
-            return False
-
-        return True
-
-    def _mark_as_synced(self, trial: "Trial"):
-        self._sync_times[trial.trial_id] = time.time()
-
-    def _local_trial_logdir(self, trial: "Trial"):
-        return trial.local_path
-
-    def _remote_trial_logdir(self, trial: "Trial"):
-        return trial.local_path
-
-    def _sync_trial_dir(
-        self, trial: "Trial", force: bool = False, wait: bool = True
-    ) -> bool:
-        if not self._enabled or trial.uses_cloud_checkpointing:
-            return False
-
-        source_ip = self._trial_ips.get(trial.trial_id, None)
-
-        if not source_ip:
-            try:
-                source_ip = trial.get_ray_actor_ip()
-            except RayActorError as e:
-                logger.error(
-                    f"Trial {trial}: An error occurred when trying to get the "
-                    f"node ip where this trial is running: {e}"
-                )
+        return super(SyncConfig, cls).__new__(cls, *args, **kwargs)
 
-            # If it still does not exist, the runner is terminated.
-            if not source_ip:
-                return False
 
-        self._trial_ips[trial.trial_id] = source_ip
-
-        if not bool(int(os.environ.get(REENABLE_DEPRECATED_SYNC_TO_HEAD_NODE, "0"))):
-            # Only log a warning for remote trials, since
-            # this only affects artifacts that are saved on worker nodes.
-            if source_ip != ray.util.get_node_ip_address():
-                if log_once("deprecated_head_node_sync"):
-                    logger.warning(_SYNC_TO_HEAD_DEPRECATION_MESSAGE)
-            return False
-
-        sync_process = self._get_trial_sync_process(trial)
-
-        # Always run if force=True
-        # Otherwise, only run if we should sync (considering sync period)
-        # and if there is no sync currently still running.
-        if not force and (not self._should_sync(trial) or sync_process.is_running):
-            return False
-
-        try:
-            sync_process.wait()
-        except TuneError as e:
-            # Errors occurring during this wait are not fatal for this
-            # checkpoint, so it should just be logged.
-            logger.error(
-                f"Trial {trial}: An error occurred during the "
-                f"checkpoint syncing of the previous checkpoint: {e}"
-            )
-        sync_process.start(
-            source_ip=source_ip,
-            source_path=self._remote_trial_logdir(trial),
-            target_ip=ray.util.get_node_ip_address(),
-            target_path=self._local_trial_logdir(trial),
-            exclude=_EXCLUDE_FROM_SYNC,
+@Deprecated
+class Syncer:
+    def __new__(cls: type, *args, **kwargs):
+        # TODO(justinvyu): Link to a custom fs user guide
+        raise DeprecationWarning(
```

Comment:
hard deprecation for Syncer interface that can no longer be used meaningfully by users